edge/pkg/metamanager/metaserver/kubernetes/storage/sqlite/store.go

```diff
@@ -99,6 +101,24 @@ func (s *store) List(ctx context.Context, key string, opts storage.ListOptions,
 		if err != nil {
 			return err
 		}
+
+		labelSet := labels.Set(unstrObj.GetLabels())
+		if !opts.Predicate.Label.Matches(labelSet) {
+			continue
+		}
+
+		// TODO: only support metadata.name & metadata.namespace
+		if _, found := opts.Predicate.Field.RequiresExactMatch("metadata.namespace"); found && unstrObj.GetNamespace() == "" {
```

Comment:
`opts.Predicate.Field.RequiresExactMatch("metadata.namespace")`
I think it is more appropriate to put this logic before the loop processing.

Reply:
Done